### PR TITLE
More database related changes

### PR DESCRIPTION
**Pull request by roonil-wazlib**
- Edit pom and connection files to get project running in production
 - Add creation of other tables (competitions, investments, participants)
 - Adjust current servlets to fetch data from db instead of returning hard coded objects
 - Add hard coded example data to database, as described in planning doc https://docs.google.com/document/d/1oOOSnXOIXO3MwaflbQMzBJA2jqB5E_tFTw8bzHVsqGs/edit?ts=5fffabfc&resourcekey=0-x1Ouq19wcIy8gic99euweg
 - hide database password in config file

### CODE REVIEW

File: project/src/main/java/com/google/sps/servlets/AuthServlet.java [hunk 1]
```diff
@@ -22,17 +22,55 @@
...
+      response.getWriter().write("Unable to successfully fetch user");
+    }
+  }
 
-    response.getWriter().println(gson.toJson(userObj));
+  private User getUser(Connection conn, String email) throws SQLException {
+    String stmt = "SELECT id, name FROM users WHERE email='" + email + "';";
+    try (PreparedStatement userStmt = conn.prepareStatement(stmt);) {
+      // Execute the statement
+      ResultSet userRs = userStmt.executeQuery();
```
Comment (texm) [inline project/src/main/java/com/google/sps/servlets/AuthServlet.java:63]: `useRs`->`users`?

File: project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java [hunk 2]
```diff
@@ -40,35 +42,66 @@
...
     HikariConfig config = new HikariConfig();
 
     log.info("I'm alive");
 
+    Config mySecrets = new Config();
+
     config.setJdbcUrl(String.format("jdbc:mysql://localhost:3306/%s", "test"));
     config.setUsername("root"); // e.g. "root", "mysql"
-    config.setPassword(REDACTED); // e.g. "my-password"
+    config.setPassword(mySecrets.databasePassword); // e.g. "my-password"
```
Comment (texm) [inline project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java:81]: Can probably remove the example comments now

File: project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java [hunk 3]
```diff
@@ -28,6 +28,8 @@
 import javax.servlet.annotation.WebListener;
 import javax.sql.DataSource;
 import java.util.logging.Logger;
+import com.google.sps.config.*;
+import com.google.common.base.Ascii;
```
Comment (texm) [inline project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java:32]: Unused import

File: project/src/main/java/com/google/sps/servlets/InvestmentServlet.java [hunk 4]
```diff
@@ -25,23 +25,71 @@
...
+        sellDateOrNull = rs.getDate(3);
+        if (sellDateOrNull == null) {
+          sellDate = 0;
+        } else {
+          sellDate = sellDateOrNull.getTime();
+        }
+        amtInvested = rs.getInt(4);
+        LOGGER.log(Level.WARNING, "GOING");
+        investments.add(Investment.create(googleSearch, investDate, sellDate, amtInvested));
+        LOGGER.log(Level.WARNING, "GOING");
```
Comment (texm) [inline project/src/main/java/com/google/sps/servlets/InvestmentServlet.java:90]: where is it going?

File: project/src/main/java/com/google/sps/servlets/AuthServlet.java [hunk 5]
```diff
@@ -22,17 +22,55 @@
...
+
 @WebServlet("/authentication")
 public class AuthServlet extends HttpServlet {
 
+  private static final Logger LOGGER = Logger.getLogger(AuthServlet.class.getName());
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    User userObj = User.create(1896,"Phoebe Khokgawe","phoebek@");
+    DataSource pool = (DataSource) request.getServletContext().getAttribute("my-pool");
```
Comment (texm) [inline project/src/main/java/com/google/sps/servlets/AuthServlet.java:41]: Maybe change `my-pool` to something more descriptive like `db-connection-pool` ?

File: project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java [hunk 6]
```diff
@@ -40,35 +42,66 @@
...
+        createUsersStatement.execute();
+      } try (PreparedStatement createCompetitionsStatement = conn.prepareStatement(CREATE_COMPETITIONS_TABLE);) {
+        createCompetitionsStatement.execute();
+      } try (PreparedStatement createParticipantsStatement = conn.prepareStatement(CREATE_PARTICIPANTS_TABLE);) {
+        createParticipantsStatement.execute();
+      } try (PreparedStatement createInvestmentsStatement = conn.prepareStatement(CREATE_INVESTMENTS_TABLE);) {
+        createInvestmentsStatement.execute();
       }
+      //get rid of this when we no longer need hard coded data
+      insertTestData(pool);
```
Comment (texm) [inline project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java:104]: Won't this create duplicate data on each startup? I would rather test data be added/removed manually.

File: project/src/main/java/com/google/sps/servlets/AuthServlet.java (hunk 1 above)
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/servlets/AuthServlet.java:63]: userRs, for user resultset. Can change to userRS?

File: project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java (hunk 6 above)
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java]: I've just been leaving it commented out, except when I change something in the database and need to drop all the tables and reinsert data.

File: project/src/main/java/com/google/sps/servlets/AuthServlet.java (hunk 5 above)
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/servlets/AuthServlet.java:41]: Done

File: project/src/main/java/com/google/sps/servlets/InvestmentServlet.java (hunk 4 above)
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/servlets/InvestmentServlet.java]: I think this is an old version

File: project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java (hunk 3 above)
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java]: Sorted

File: project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java (hunk 2 above)
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/database/ConnectionPoolContextListener.java]: Done

File: project/src/main/java/com/google/sps/servlets/AuthServlet.java (hunk 1 above)
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/servlets/AuthServlet.java]: Modified to rs, seems to be convention.

**Review by lin8231 — APPROVED** [review]

**Review by texm — APPROVED** [review]

**Pull request merged**